File: leetcode/golang/LCR_144_er_cha_shu_de_jing_xiang_lcof.go
type TreeNode struct {
         Val int
         Left *TreeNode
         Right *TreeNode
}

   
func dfs(root *TreeNode) {
    if root == nil {
        return
    }
    root.Left, root.Right = root.Right, root.Left
    dfs(root.Left)
    dfs(root.Right)
}
func mirrorTree(root *TreeNode) *TreeNode {
    dfs(root)
    return root
}

File: leetcode/c++/LCR_144_er_cha_shu_de_jing_xiang_lcof.cpp
// https://leetcode.cn/problems/er-cha-shu-de-jing-xiang-lcof/description

#include <bits/stdc++.h>
using namespace std;

struct TreeNode {
      int val;
      TreeNode *left;
      TreeNode *right;
      TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution {
public:
    void reverse(TreeNode* root) {
        if (root == NULL)
            return;
        TreeNode* node = root->left;
        root->left = root->right;
        root->right = node;
        reverse(root->left);
        reverse(root->right);
    }
    TreeNode* mirrorTree(TreeNode* root) {
        reverse(root);
        return root;
    }
};

// 时间复杂度O(n)，空间复杂度O(n)